parm/land/letkfoi/letkfoi.yaml

```diff
@@ -56,5 +56,5 @@ output increment:
   filetype: fms restart
   filename_sfcd: '{{ current_cycle | to_fv3time }}.sfc_data.nc'
   filename_cplr: '{{ current_cycle | to_fv3time }}.coupler.res'
-  state variables: [snwdph,vtype,slmsk]
+  state variables: [$(SNOWDEPTHVAR),vtype,slmsk]
```

Comment:
@jiaruidong2017 in case it is lost, I think Rahul would like this to be `[  {{ SNOWDEPTHVAR }} ,` or something close to that

Reply:
@CoryMartin-NOAA I tried using the suggested format ` {{ SNOWDEPTHVAR }}`, but it caused the failure in the ctest `test_gdasapp_genYAML_land_run` (see the conversations below)

Reply:
hmm, I wonder if it is because it is part of a list.

Reply:
can you do: `['{{ SNOWDEPTHVAR }}', vtype, slmsk]`? Does that work?

Reply:
I will try this after hera is back. Thanks.

Reply:
Make sure `SNOWDEPTHVAR` is a member of `self.task_config` prior to https://github.com/NOAA-EMC/global-workflow/blob/188344a01297a8dede34f2f3735536b94b9ab843/ush/python/pygfs/task/land_analysis.py#L197C39-L197C81
It should be fine a `[{{ SNOWDEPTHVAR }}, ...]` No need for `' '`

Reply:
The proposed https://github.com/NOAA-EMC/global-workflow/pull/1687 defined the variable `SNOWDEPTHVAR` as a member of `self.task_config`. The failure came from the ctest in GDASApp rather than the tests in G-W. I will try to figure out it following your suggestions after hera is back online. Thanks.

Reply:
@aerorahul I tried using `state variables: [{{ SNOWDEPTHVAR }},vtype,slmsk]`, but it failed. If I tried to use `state variables: '{{ SNOWDEPTHVAR }},vtype,slmsk'`, the ctest passed. 
@CoryMartin-NOAA Can I use `state variables: '{{ SNOWDEPTHVAR }},vtype,slmsk'` here?

Reply:
what about `state variables: ['{{ SNOWDEPTHVAR }}',vtype,slmsk]`

Reply:
@CoryMartin-NOAA It works.

Reply:
The output is below:

```
      state variables:
      - '{{ SNOWDEPTHVAR }}'
      - vtype
      - slmsk
```

Reply:
I still don't know why the `' '` is needed around `{{ SNOWDEPTHVAR }}`.  @jiaruidong2017 can you point me to the ctest that is failing and how this yaml is being generated?

Reply:
The issue is the test is using https://github.com/NOAA-EMC/GDASApp/blob/develop/ush/ufsda/genYAML.py.  This function is not the same as the one used in the global-workflow. global-workflow is using jinja templates. https://github.com/NOAA-EMC/GDASApp/blob/develop/ush/ufsda/genYAML.py uses templates from pygw, but they are not jinja templates.
So even if this renders correctly, it is not really a test of how this yaml will be rendered in global-workflow.

Reply:
@aerorahul The ctests are below:

```
test_gdasapp_genYAML_land_prep
test_gdasapp_genYAML_land_run
```

Reply:
When I use `state variables: [$(SNOWDEPTHVAR),vtype,slmsk]`, the out put is below:

```
      state variables:
      - snodl
      - vtype
      - slmsk
```

Reply:
For above output, I defined `SNOWDEPTHVAR: 'snodl'` in `genYAML_prep_land.sh`.